types/block_finality_test.go

```diff
@@ -282,3 +287,51 @@ func TestBlockNumberFinalityJSONSchema(t *testing.T) {
 	require.Equal(t, "string", schema.Type)
 	require.Equal(t, "BlockNumberFinality", schema.Title)
 }
+
+func TestBlockNumberFinality_BlockHeader(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("Success with offset", func(t *testing.T) {
+		mockClient := mocks.NewBaseEthereumClienter(t)
+		blockFinality := BlockNumberFinality{Block: Finalized, Offset: -5}
+
+		finalizedHeader := &types.Header{Number: big.NewInt(100)}
+		offsetHeader := &types.Header{Number: big.NewInt(95)}
+
+		mockClient.On("HeaderByNumber", ctx, big.NewInt(int64(rpc.FinalizedBlockNumber))).Return(finalizedHeader, nil).Once()
+		mockClient.On("HeaderByNumber", ctx, big.NewInt(95)).Return(offsetHeader, nil).Once()
+
+		result, err := blockFinality.BlockHeader(ctx, mockClient)
+		require.NoError(t, err)
+		require.Equal(t, offsetHeader, result)
+	})
+
+	t.Run("Error on first call", func(t *testing.T) {
+		mockClient := mocks.NewBaseEthereumClienter(t)
+		blockFinality := BlockNumberFinality{Block: Latest, Offset: 0}
+
+		testErr := fmt.Errorf("first call error")
+		mockClient.On("HeaderByNumber", ctx, (*big.Int)(nil)).Return(nil, testErr).Once()
```

Comment:
Use the expecter, as it is compile time safe. The same comment applies for other places as well.

```suggestion
		mockClient.EXPECT().HeaderByNumber(ctx, (*big.Int)(nil)).Return(nil, testErr).Once()
```

Reply:
fixed: 8bbb797d15b7a1d63a88741bcb87348acc81496d